app/root.tsx

```diff
@@ -81,27 +76,31 @@ export function useRootLoaderData() {
   return useRouteLoaderData('root') as SerializeFrom<typeof loader>
 }
 
-// 404 Catch
-export function CatchBoundary() {
-  const caught = useCatch()
+export function ErrorBoundary() {
+  const error = useRouteError()
 
-  if (caught.status === 404) {
-    return <NotFound />
+  let message = 'Something went wrong'
+
+  if (isRouteErrorResponse(error)) {
+    if (error.status === 404) {
+      message = '404 Not Found'
+    }
   }
-}
 
+  return (
+    <Layout>
+      <div className="flex h-full w-full items-center justify-center">
+        <h1 className="text-2xl">{message}</h1>
+      </div>
+    </Layout>
+  )
+}
 const queryClient = new QueryClient()
 
-export const Layout = ({
-  children,
-  theme,
-}: {
-  children: React.ReactNode
-  theme?: string
-}) => (
+const Layout = ({ children, theme }: { children: React.ReactNode; theme?: string }) => (
```

Comment:
Here's a fun one. When `Layout` is exported it nests the layout inside itself. Unsure why but no longer need to export now I'm not using it elsewhere.

Reply:
`Layout` is a special export in root, haha. If you name it something else it would work as expected.

https://github.com/remix-run/remix/pull/8709